postprocessing/land_utils/scatter-ncc.F90

```diff
@@ -289,26 +289,28 @@ program combine_res
            if(has_records) then  ! record variable
               start(:) = 1; nread(:) = 1; nwrite(:) = 1
               start(1) = tlev
-              __NF_ASRT__(nfu_get_compressed_var_r8n(in_ncid,varname,buffer,mask,start=start,count=nread))
+              CHECK_NF_ERRSTAT(nfu_get_compressed_var_r8n(in_ncid,varname,buffer,mask,start=start,count=nread))
                  do n = 1, nfiles_out
-                    __NF_ASRT__(nfu_put_vara_r8(out_ncid(n),varname,start, nwrite,buffer))
+                    CHECK_NF_ERRSTAT(nfu_put_vara_r8(out_ncid(n),varname,start, nwrite,buffer))
                  enddo
            else
-              __NF_ASRT__(nfu_get_compressed_var_r8n(in_ncid,varname,buffer,mask))
+              CHECK_NF_ERRSTAT(nfu_get_compressed_var_r8n(in_ncid,varname,buffer,mask))
               if (count(mask)>0) then
                  do n = 1, nfiles_out
-                    __NF_ASRT__(nfu_put_var_r8(out_ncid(n),varname,pack(buffer,mask)))
+                    CHECK_NF_ERRSTAT(nfu_put_var_r8(out_ncid(n),varname,pack(buffer,mask)))
                  enddo
               endif
            endif
         endif
         deallocate(buffer,mask)
      enddo
   enddo
-  __NF_ASRT__(nf_close(in_ncid))
+  CHECK_NF_ERRSTAT(nf_sync(in_ncid))
+  CHECK_NF_ERRSTAT(nf_close(in_ncid))
```

Comment:
The descriptor `in_ncid` is associated with a file opened with `NF_NOWRITE` and doesn't need to sync prior to the close.